tests/framework-tests/core/js/DataBindingTests.js

```diff
@@ -2076,6 +2076,170 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+        jqUnit.expect(fluid.keys(testCases).length);
+        fluid.each(testCases, function (newModel, caseName) {
+            that.sub.applier.fireChangeRequest({path: "", type: "DELETE"});
+            that.sub.applier.change("", newModel);
+            jqUnit.assertDeepEq("The new model value has been relayed properly for " + caseName, newModel, that.relayedModelValue);
+        });
+    };
+
+    fluid.tests.fluid5585.verifyForwardRelay = function (that, testCases) {
+        jqUnit.expect(fluid.keys(testCases).length);
+        fluid.each(testCases, function (newModel, caseName) {
+            that.applier.fireChangeRequest({path: "subModel", type: "DELETE"});
+            that.applier.change("subModel", newModel);
+            jqUnit.assertDeepEq("The new model value has been relayed properly for " + caseName, newModel, that.sub.model);
+        });
+    };
+
+    fluid.tests.fluid5585.verifyForwardPartialDeleteRelay = function (that) {
+        jqUnit.expect(1);
+        that.applier.fireChangeRequest({path: "subModel.b", type: "DELETE"});
+        jqUnit.assertDeepEq("The new model value has been relayed properly for partial deletion", {a: true}, that.sub.model);
+    };
+
+    fluid.tests.fluid5585.verifyBackwardPartialDeleteRelay = function (that) {
+        jqUnit.expect(1);
+        that.sub.applier.fireChangeRequest({path: "b", type: "DELETE"});
+        jqUnit.assertDeepEq("The new model value has been relayed properly for partial deletion", {a: true}, that.model.subModel);
+    };
+
+    fluid.tests.fluid5585.runOneConfiguration = function (gradeName, verifyFunc) {
+        jqUnit.test("FLUID-5585 - " + gradeName + " with " + verifyFunc + ": model relay for removing all or part of source value nodes", function () {
+            jqUnit.expect(2);
+            var that = fluid.tests.fluid5585.root({
+                components: {
+                    sub: {
+                        options: {
+                            gradeNames: gradeName
+                        }
+                    }
+                }
+            });
+            jqUnit.assertDeepEq(gradeName + ": the initial model value on the source component is set correctly", that.initialModelValue, that.sub.model);
+            jqUnit.assertDeepEq(gradeName + ": the initial model value on the target component is set correctly", that.initialModelValue, that.model.subModel);
+            fluid.invokeGlobalFunction(verifyFunc, [that, fluid.tests.fluid5585fixtures]);
+        });
+    };
+
+    fluid.tests.fluid5585.fixtureFuncs = [
+        "fluid.tests.fluid5585.verifyBackwardRelay",
+        "fluid.tests.fluid5585.verifyForwardRelay",
+        "fluid.tests.fluid5585.verifyForwardPartialDeleteRelay",
+        "fluid.tests.fluid5585.verifyBackwardPartialDeleteRelay"
+    ];
+
+    fluid.each(fluid.tests.fluid5585.fixtureFuncs, function (fixtureFunc) {
+        fluid.tests.fluid5585.runOneConfiguration("fluid.tests.fluid5585.explicitRelay", fixtureFunc);
```

Comment:
Line 2206 and 2207 are identical. Did you mean to have one of them test `fluid.tests.fluid5585.implicitRelay`?